### PR TITLE
Groups resources tags

### PR DESCRIPTION
**Pull request by leoalfonso14**
### About
This PR makes the process of loading the group resources when adding them to a group faster by query them as TaggedItems when a category is selected.

_TODO: Tag all resources so they show up before merging into main_
### Test Instructions
1. Open the testing link and log in.
2. Click on your profile picture on the top right and go to your groups.
3. Open any of your groups, click on manage group and try to add a resource.
4. Compare the difference of loading speed compared to live website.
5. Also check that it allows to add and remove resources, and that if a resource is already added to the group resources that it does not show as available.

### Screenshots
Screenshot 2023-09-22 at 2 05 13 PM

### Closes Tickets
[CFDP-2612](https://christfellowshipchurch.atlassian.net/browse/CFDP-2612)

[CFDP-2612]: https://christfellowshipchurch.atlassian.net/browse/CFDP-2612?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dzwood** [comment]
Great job on this!! Next steps is to reach out to Pam and make sure that all the Resource Items are tagged properly😁

**Review by codingwithlis — DISMISSED** [review]

**Review by codingwithlis — APPROVED** [review]

**Review by dzwood — APPROVED** [review]
Awesome job on this @leoalfonso14 🎉 We'll just wait for the team to finishing tagging all the resources in Rock before merging😄

File: components/GroupManage/AddResourceContent.js
Comment (dzwood) [inline components/GroupManage/AddResourceContent.js:1]: @leoalfonso14 Could we add a loading state for the Categories list and each of the Tags? Sometimes when I click on a Tag the items aren't loaded yet and displays an empty dropdown:
image

We should try to disable the dropdown until the items are done loading. Also we may want to consider removing empty categories.

**Review by dzwood — CHANGES_REQUESTED** [review]

File: components/GroupManage/AddResourceContent.js
Comment (leoalfonso14) [reply components/GroupManage/AddResourceContent.js:1]: I will look into it, thank you!

**Review by codingwithlis — APPROVED** [review]

**Review by dzwood — APPROVED** [review]
This look awesome!! Great job on those loading states too🎉

**Pull request merged**